src/isar/models/mission.py

```diff
@@ -47,3 +70,6 @@ def __post_init__(self) -> None:
 
         if self.metadata is None:
             self.metadata = MissionMetadata(mission_id=self.id)
+
+        if self.iterator is None:
+            self.iterator = iter(self.tasks)
```

Comment:
This looks a bit strange here. I assume it is done like this because we previously set tasks on the mission after the mission was instantiated. But now that this is changed (see other comment), I think we can move it to a post_init as it is done in `Task`

Reply:
It is already in the `__post_init__` function I believe. I'll rename it to also be internal.